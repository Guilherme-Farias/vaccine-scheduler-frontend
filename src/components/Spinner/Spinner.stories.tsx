import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Spinner from '.';

export default {
  title: 'Spinner',
  component: Spinner,
  argTypes: {
    color: {
      description: 'Change the color',
      defaultValue: 'gray-400',
      control: {
        type: 'select',
      },
    },
  },
} as ComponentMeta<typeof Spinner>;

const Template1: ComponentStory<typeof Spinner> = args => <Spinner {...args} />;

const Template2: ComponentStory<typeof Spinner> = args => (
  <div>
    <Spinner {...args} color="white" />
    <Spinner {...args} color="gray-50" />
    <Spinner {...args} color="gray-100" />
    <Spinner {...args} color="gray-200" />
    <Spinner {...args} color="gray-300" />
    <Spinner {...args} color="gray-400" />
    <Spinner {...args} color="gray-500" />
    <Spinner {...args} color="gray-600" />
    <Spinner {...args} color="gray-700" />
    <Spinner {...args} color="gray-800" />
    <Spinner {...args} color="gray-900" />
    <Spinner {...args} color="black" />
    <Spinner {...args} color="primary" />
    <Spinner {...args} color="secondary" />
    <Spinner {...args} color="tertiary" />
    <Spinner {...args} color="red" />
    <Spinner {...args} color="green" />
    <Spinner {...args} color="blue" />
    <Spinner {...args} color="yellow" />
  </div>
);

export const Default = Template1.bind({});
Default.args = {};

export const TableOfVariants = Template2.bind({});
TableOfVariants.args = {};
