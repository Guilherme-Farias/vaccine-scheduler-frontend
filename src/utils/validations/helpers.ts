import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
