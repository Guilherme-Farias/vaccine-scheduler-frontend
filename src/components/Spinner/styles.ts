import styled, { css } from 'styled-components/macro';

import { SpinnerProps } from './Spinner';

export const Container = styled.div<SpinnerProps>`
  ${({ theme, color }) => css`
    --spinner-size: ${theme.spacings['3xl']};
    display: inline-block;
    width: var(--spinner-size);
    height: var(--spinner-size);
    vertical-align: -0.125em;
    border: ${theme.spacings['2xs']} solid ${theme.colors[color!]};
    border-right-color: transparent;
    border-radius: 50%;
    animation: 0.75s linear infinite rotate;

    @keyframes rotate {
      100% {
        transform: rotate(360deg);
      }
    }
  `}
`;
