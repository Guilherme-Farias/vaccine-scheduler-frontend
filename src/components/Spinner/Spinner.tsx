import VisuallyHidden from 'components/VisuallyHidden';
import React from 'react';
import theme from 'styles/theme';
import * as S from './styles';

type ThemeColors = typeof theme.colors;

export type SpinnerProps = {
  color?: keyof Omit<ThemeColors, 'mainBg' | 'minorBg'>;
};

const Spinner: React.FC<SpinnerProps> = ({
  color = 'gray-400',
}: SpinnerProps) => (
  <S.Container role="status" color={color}>
    <VisuallyHidden>Carregando...</VisuallyHidden>
  </S.Container>
);

export default Spinner;
