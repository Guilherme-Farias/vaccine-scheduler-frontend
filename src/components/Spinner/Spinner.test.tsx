import { render, screen } from 'utils/tests/helpers';

import Spinner from '.';

describe('<Spinner />', () => {
  it('should render the Spinner', () => {
    const { container } = render(<Spinner />);
    expect(screen.getByRole('status')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
