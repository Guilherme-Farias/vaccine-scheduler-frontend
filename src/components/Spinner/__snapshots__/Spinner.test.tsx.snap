// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render the Spinner 1`] = `
.c1 {
  position: absolute;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
}

.c0 {
  --spinner-size: 2rem;
  display: inline-block;
  width: var(--spinner-size);
  height: var(--spinner-size);
  vertical-align: -0.125em;
  border: 0.25rem solid #9ca3af;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: 0.75s linear infinite rotate;
  animation: 0.75s linear infinite rotate;
}

<div
  class="c0"
  color="gray-400"
  role="status"
>
  <div
    class="c1"
  >
    Carregando...
  </div>
</div>
`;
